{"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/vapor\/debuggableerror\/readablename-5uonz"]}],"metadata":{"roleHeading":"Type Property","symbolKind":"property","modules":[{"name":"Vapor"}],"title":"readableName","role":"symbol","extendedModule":"Vapor","externalID":"s:5Vapor15DebuggableErrorPAAE12readableNameSSvpZ","fragments":[{"text":"static","kind":"keyword"},{"text":" ","kind":"text"},{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"readableName","kind":"identifier"},{"text":": ","kind":"text"},{"text":"String","preciseIdentifier":"s:SS","kind":"typeIdentifier"}]},"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor\/DebuggableError\/readableName-5uonz"},"sections":[],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["Linux"],"tokens":[{"text":"static","kind":"keyword"},{"text":" ","kind":"text"},{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"readableName","kind":"identifier"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":" { ","kind":"text"},{"text":"get","kind":"keyword"},{"kind":"text","text":" }"}],"languages":["swift"]}]},{"kind":"content","content":[{"anchor":"discussion","level":2,"text":"Discussion","type":"heading"},{"name":"Note","style":"note","content":[{"inlineContent":[{"type":"text","text":"For example, an error named "},{"code":"FooError","type":"codeVoice"},{"type":"text","text":" will have the"},{"type":"text","text":" "},{"code":"readableName","type":"codeVoice"},{"type":"text","text":" "},{"code":"\"Foo Error\"","type":"codeVoice"},{"type":"text","text":"."}],"type":"paragraph"}],"type":"aside"}]}],"abstract":[{"text":"A readable name for the error’s Type. This is usually","type":"text"},{"text":" ","type":"text"},{"text":"similar to the Type name of the error with spaces added.","type":"text"},{"text":" ","type":"text"},{"text":"This will normally be printed proceeding the error’s reason.","type":"text"}],"schemaVersion":{"patch":0,"minor":3,"major":0},"kind":"symbol","hierarchy":{"paths":[["doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor","doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor\/DebuggableError","doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor\/DebuggableError\/readableName-55ap2"]]},"references":{"doc://co.pointfree.VaporRouting/documentation/Vapor/DebuggableError":{"navigatorTitle":[{"text":"DebuggableError","kind":"identifier"}],"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor\/DebuggableError","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"DebuggableError","kind":"identifier"}],"role":"symbol","abstract":[{"type":"codeVoice","code":"Debuggable"},{"type":"text","text":" provides an interface that allows a type"},{"type":"text","text":" "},{"type":"text","text":"to be more easily debugged in the case of an error."}],"title":"DebuggableError","type":"topic","url":"\/documentation\/vapor\/debuggableerror","kind":"symbol"},"doc://co.pointfree.VaporRouting/documentation/Vapor/DebuggableError/readableName-55ap2":{"type":"topic","title":"readableName","role":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor\/DebuggableError\/readableName-55ap2","kind":"symbol","fragments":[{"kind":"keyword","text":"static"},{"kind":"text","text":" "},{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"readableName"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"defaultImplementations":1,"abstract":[{"type":"text","text":"A readable name for the error’s Type. This is usually"},{"type":"text","text":" "},{"type":"text","text":"similar to the Type name of the error with spaces added."},{"type":"text","text":" "},{"type":"text","text":"This will normally be printed proceeding the error’s reason."}],"url":"\/documentation\/vapor\/debuggableerror\/readablename-55ap2","required":true},"doc://co.pointfree.VaporRouting/documentation/Vapor/DebuggableError/readableName-5uonz":{"type":"topic","title":"readableName","role":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor\/DebuggableError\/readableName-5uonz","kind":"symbol","fragments":[{"kind":"keyword","text":"static"},{"kind":"text","text":" "},{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"readableName"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"abstract":[{"type":"text","text":"A readable name for the error’s Type. This is usually"},{"type":"text","text":" "},{"type":"text","text":"similar to the Type name of the error with spaces added."},{"type":"text","text":" "},{"type":"text","text":"This will normally be printed proceeding the error’s reason."}],"url":"\/documentation\/vapor\/debuggableerror\/readablename-5uonz"},"doc://co.pointfree.VaporRouting/documentation/Vapor":{"type":"topic","abstract":[],"role":"collection","url":"\/documentation\/vapor","kind":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/Vapor","title":"Vapor"}}}